### PR TITLE
Update GO_TOOLCHAIN string to Label and add toolchain param to one more action

### PR DESCRIPTION
**Pull request by kotlaja**
What type of PR is this?
Other - Migration

What does this PR do? Why is it needed?
Because of the relative paths, we'll need GO_TOOLCHAIN as a Label, not a String. I saw this error by building `@go_default_sdk//...` from tests/bcr workspace.

Since it's not easy to see which actions are affected by these changes (sometimes it depends on projects which are using them), I'm updating these actions until I see green Bazel downstream projects. Before that I build the whole repo manually with the incompatible flag, but in some projects that is not enough (combinations of flags can affect this, executable/tools as variables from other projects, etc).

Which issues(s) does this PR fix?
Please check [official documentation for Automatic Exec Groups](https://bazel.build/extending/auto-exec-groups) for additional info.

### CODE REVIEW

**Comment by kotlaja** [comment]
@fmeum Can you please check this one too? :)

**Comment by fmeum** [comment]
@kotlaja Looks good to me, but there is a failure with Bazel 5.4.0. Could you take a look?

**Comment by kotlaja** [comment]
Ahh, Bazel 5.4.0 supports toolchain registration on a rule only as a String, not as a Label ([see String parsing here](https://github.com/bazelbuild/bazel/blob/release-5.4.0/src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkRuleClassFunctions.java#L627)).

I've used `str` func for this. PTAL. :)

File: go/private/common.bzl [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GO_TOOLCHAIN = "@io_bazel_rules_go//go:toolchain"
+GO_TOOLCHAIN = Label("@io_bazel_rules_go//go:toolchain")
```
Comment (fmeum) [inline go/private/common.bzl:15]: How about applying `str` here instead?

Comment (kotlaja) [reply go/private/common.bzl]: I also thought of adding one more constant as a string, but let's go with your suggestion. Thanks! :)

**Review by fmeum — APPROVED** [review]

**Comment by fmeum** [comment]
CI is still failing with a very obscure failure.

**Comment by kotlaja** [comment]
Seems that Bazel 5.4. has issues with `str(Label(toolchain_string))` when registering toolchains on rules. The error is related to modified string of a toolchain which must be mapped 1-1 (changing it to label and then to a string changes that) or otherwise toolchain resolution doesn't find it as a registered one. I've separated it now into a Label and a String and updated toolchain parameters.

**Review by fmeum — APPROVED** [review]

**Comment by kotlaja** [comment]
Thanks @fmeum for the quick responses and all the help! :)
By checking rules_go with multiple rules today, it seems that these changes are sufficient for the AEGs migration. 

I know we had a release 4 days ago, but I'm interested when can we expect the next one? 
I'll CC also @tyler-french here - Hi Tyler! :)

**Pull request merged**